packages/backend/src/secret.ts

```diff
@@ -27,6 +27,10 @@ import { BackendSecretFetcherFactory } from './engine/backend-secret/backend_sec
  *          ```
  */
 export const secret = (name: string): BackendSecret => {
+  // @ts-expect-error - this is a runtime check to prevent client-side usage
```

Comment:
could you try https://stackoverflow.com/questions/73261451/how-to-detect-if-code-runs-in-browser-in-typescript ?